tests/test_standard_loader.py

```diff
@@ -0,0 +1,278 @@
+        bundle['data_objects'] = {minimal_file_info_guid: incomplete_file_info}
+        self.assertRaises(ParseError, self.loader._parse_bundle, bundle)
+        incomplete_file_info.update(minimal_file_info)
+        self.assertTrue(type(self.loader._parse_bundle(bundle)) == ParsedBundle)
+
+    def test_parse_all_bundles(self):
+        """
+        Try and parse multiple bundles.
+
+        The sample file is just an already transformed standard format version of the TopMed public 107
+        """
+        self.assertRaises(ParseError, self.loader._parse_all_bundles, 'not a json')
+
+        input_json = load_json_from_file(str(TEST_DATA_PATH / 'multiple_bundles.json'))
+        self.loader._parse_all_bundles(input_json)
+        self.assertEqual(len(self.loader.bundles_parsed), len(input_json))
+        self.assertEqual(len(self.loader.bundles_failed_unparsed), 0)
+
+    # TODO add some tests for credentials and stuff so that we get nice error messages
+
+    def _test_loading_bundles_dict(self, bundles: typing.List[dict]):
+        # Nothing should have been processed at this point
+        self.assertEqual(len(self.loader.bundles_parsed), 0)
+        self.assertEqual(len(self.loader.bundles_loaded), 0)
+        self.assertEqual(len(self.loader.bundles_failed_unparsed), 0)
+        self.assertEqual(len(self.loader.bundles_failed_parsed), 0)
+
+        self.loader.load_all_bundles(bundles)
+
+        self.assertEqual(len(self.loader.bundles_loaded), len(bundles))
+        self.assertEqual(len(self.loader.bundles_parsed), len(bundles))
+        self.assertEqual(len(self.loader.bundles_failed_unparsed), 0)
+        self.assertEqual(len(self.loader.bundles_failed_parsed), 0)
+
+    def _make_minimal_bundle(self, parsed=True):
+        bundle_uuid = str(uuid.uuid4())
+        metadata_dict = {'some': 'stuff', 'more': 'stuff'}
+
+        file_contents = io.BytesIO(b'This is a very important file.\n'
+                                   b'The content is pretty important but is self referential\n')
+        filename = f'minimal-file-{uuid.uuid4()}'
+        file_uuid = str(uuid.uuid4())
+        file_guid = f'dg.405/{file_uuid}'
+        file_version = tz_utc_now()
+        # we only need one URL, but it needs to be valid
+        cloud_urls = [f's3://{self.bucket_name}/{filename}']
+
+        # do the cloud upload
+        file = self.s3.Object(self.bucket_name, filename)
+        file.put(Body=file_contents)
+        file.Acl().put(ACL='public-read')
+
+        if parsed:
+            return ParsedBundle(bundle_uuid,
+                                metadata_dict,
+                                [ParsedDataFile(filename, file_uuid, cloud_urls, bundle_uuid, file_guid, file_version)])
+        else:
+            minimal = {
+                'data_bundle': {
+                    'id': bundle_uuid,
+                    'data_object_ids': file_guid,
+                    'created': tz_utc_now(),  # FIXME is this necessary?????
```

Comment:
I think `created` is needed. What is `this` referring to?

Reply:
Just left over development cruft. Thanks for catching it